### ISSUE TITLE
Support debug-compact serializer?

### ISSUE THREAD

**Issue by tisonkun** [mitsuhiko/insta#533]
We now have `assert_debug_snapshot` which uses  `"{:#?}"` a.k.a. pretty print.

I wonder if it's reasonable to add a `assert_compact_debug_snapshot` use `"{:?}"` so that we may save some lines.

**Comment by tisonkun** [comment]
Currently, it's a bit wordy:

```rust
    assert_debug_snapshot!(
        candidates,
        @r###"
        [
            0,
        ]
        "###
    );
```

using ron reduce two spaces in the indent, but doesn't help too much.

```rust
assert_ron_snapshot!(
        candidates,
        @r###"
        [
          0,
        ]
        "###
    );
```

Of course, I can use `assert_snapshot!(format!("{candidates:?}"), @r"[0]");`, but it seems insta try to provide convenient serializer out of the box.

**Comment by max-sixty** [comment]
Waiting on review in #514 !
